### PR TITLE
IGNITE-16549 .NET: Add IgniteClientConfiguration.SendServerExceptionStackTraceToClient

### PR DESCRIPTION
**Pull request by ptupitsyn**
* Propagate SendServerExceptionStackTraceToClient to .NET
* Java fix: Include stack trace for Compute errors
* Java fix: Include stack trace even when cause is null

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java [hunk 1]
```diff
@@ -125,8 +125,7 @@
             msg = sqlState + ": " + msg;
         }
 
-        if ((X.getCause(e) != null || !X.getSuppressedList(e).isEmpty())
-                && ctx.kernalContext().sqlListener().sendServerExceptionStackTraceToClient())
+        if (ctx.kernalContext().sqlListener().sendServerExceptionStackTraceToClient())
```
Comment (isapego) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java:128]: Should we preserve `!X.getSuppressedList(e).isEmpty()` logic here?

**Review by isapego — APPROVED** [review]
Looks good overall. See a single comment I have.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java (hunk 1 above)
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java:128]: I'm not sure why. The option is called `sendServerExceptionStackTraceToClient`, so we just send the stack trace whatever it is, when enabled.

With those conditions in place I tried a few things like "disabled compute", "exception in compute task" and never got it to work.

Comment (isapego) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java:128]: Ok, your solution is reasonable then

**Pull request merged**